identity/service.go

```diff
@@ -195,43 +221,108 @@ func (s *Service) ProposeVote(v *ProposeVote) (network.Message, onet.ClientError
 	}
 
 	// Propagate the vote
-	_, err := s.propagateConfig(sid.Root.Roster, v, propagateTimeout)
+	_, err := s.propagateData(sid.SCRoot.Roster, v, propagateTimeout)
 	if err != nil {
 		return nil, onet.NewClientErrorCode(ErrorOnet, cerr.Error())
 	}
-	if len(sid.Votes) >= sid.Latest.Threshold ||
-		len(sid.Votes) == len(sid.Latest.Device) {
+	votesCnt := len(sid.Proposed.Votes)
+	if votesCnt >= sid.Latest.Threshold ||
+		votesCnt == len(sid.Latest.Device) {
 		// If we have enough signatures, make a new data-skipblock and
 		// propagate it
 		log.Lvl3("Having majority or all votes")
 
 		// Making a new data-skipblock
 		log.Lvl3("Sending data-block with", sid.Proposed.Device)
-		reply, cerr := s.skipchain.StoreSkipBlock(sid.Data, nil, sid.Proposed)
+		reply, cerr := s.skipchain.StoreSkipBlock(sid.SCData, nil, sid.Proposed)
 		if cerr != nil {
 			return nil, cerr
 		}
 		_, msg, _ := network.Unmarshal(reply.Latest.Data)
-		log.Lvl3("SB signed is", msg.(*Config).Device)
+		log.Lvl3("SB signed is", msg.(*Data).Device)
 		usb := &UpdateSkipBlock{
 			ID:     v.ID,
 			Latest: reply.Latest,
 		}
-		_, err = s.propagateSkipBlock(sid.Root.Roster, usb, propagateTimeout)
+		_, err = s.propagateSkipBlock(sid.SCRoot.Roster, usb, propagateTimeout)
 		if err != nil {
 			return nil, onet.NewClientErrorCode(ErrorOnet, cerr.Error())
 		}
-		return &ProposeVoteReply{sid.Data}, nil
+		return &ProposeVoteReply{sid.SCData}, nil
 	}
 	return nil, nil
 }
 
+// VerifyBlock makes sure that the new block is legit. This function will be
+// called by the skipchain on all nodes before they sign.
+func (s *Service) VerifyBlock(sbID []byte, sb *skipchain.SkipBlock) bool {
+	// Putting it all in a function for easier error-printing
+	err := func() error {
+		if sb.Index == 0 {
+			log.Lvl4("Always accepting genesis-block")
+			return nil
+		}
+		_, dataInt, err := network.Unmarshal(sb.Data)
+		if err != nil {
+			return errors.New("got unknown packet")
+		}
+		data, ok := dataInt.(*Data)
```

Comment:
`ok` checks if it's really a `Data`-block. The `network.Unmarshal` will return an initialised block anyway, so I don't see how I could get a `nil` here. Perhaps I'm missing something?